server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1368 @@
+                        binaryId: file.binary?.file.id
+                    , cb
+        , callback
+
+
+
+displayData = (requiredFields, entries, data, next) ->
+    K.logger.info "display data"
+    K.logger.info JSON.stringify entries, null, 2
+    K.logger.info JSON.stringify data, null, 2
+
+    next()
+
+# Konnector
+K = module.exports = require('../lib/base_konnector').createNew
+    name: 'EDF'
+    slug: 'edf'
+    description: 'konnector description edf'
+    vendorLink: 'https://particulier.edf.fr/fr'
+
+    fields:
+        email: 'text'
+        password: 'password'
+        folderPath: 'folder'
+
+        # TODO : get one edeliaClientId: 'text'
+
+    models: [Client, Contract, PaymentTerms, Home, ConsumptionStatement, Bill]
+
+    # Define model requests !
+    init: (callback) ->
+        async.each [Client, Contract, PaymentTerms, Home, ConsumptionStatement]
+        , (docType, cb) ->
+            docType.defineRequest 'all', cozydb.defaultRequests.all, cb
+        , (err) ->
+            if err
+                @logger.error err
+
+    fetchOperations: [
+        prepareEntries
+
+        getEDFToken
+        fetchListerContratClientParticulier
+        fetchVisualiserPartenaire
+        fetchVisualiserAccordCommercial
+        fetchVisualiserCalendrierPaiement
+        fetchRecupereDocumentContractuelListx
+        fetchVisualiserHistoConso
+
+        # Deactivate Edelia, while clientId isn't ready.
+        #fetchEdeliaToken
+        #fetchEdeliaProfile
+        #fetchEdeliaMonthlyElecConsumptions
+        #fetchEdeliaSimilarHomeYearlyElecComparisions
+        #fetchEdeliaElecIndexes
+        #fetchEdeliaMonthlyGasConsumptions
+        #fetchEdeliaSimilarHomeYearlyGasComparisions
+        #fetchEdeliaGasIndexes
+
+        updateOrCreate logger, Client, ['clientId', 'vendor']
+        updateOrCreate logger, Contract, ['number', 'vendor']
+        updateOrCreate logger, PaymentTerms, ['vendor', 'clientId']
+        updateOrCreate logger, Home, ['pdl']
+        updateOrCreate logger, ConsumptionStatement, ['contractNumber',
+            'statementType', 'statementReason', 'statementCategory', 'start']
+        updateOrCreate logger, Bill, ['vendor', 'number']
+        saveMissingBills
+        #displayData
+    ]
+
+# Helpers
+
+_extend = (a, b) ->
+    for k, v of b
+        if v?
+            a[k] = v
+    return a
+
+getF = (node, fields...) ->
+    try
+        for field in fields
+            node = node[field][0]
+    catch e
+        return null
+
+    return node
+
+translate = (dict, name) ->
+    if name of dict
+        return dict[name]
+    return name
+
+edfRequestPost = (path, body, callback) ->
+    K.logger.debug "called edfRequestPost"
+    xmlBody = builder.buildObject body
+    request
+        url: 'https://rce-mobile.edf.com' + path
+        method: 'POST'
+        # TODO : fix SSL3_GET_RECORD:wrong version number
+        #agentOption: securityOptions: 'TLSv1_method'
```

Comment:
True, I can remove these line.
Actually, I get sometimes the error `SSL3_GET_RECORD:wrong version number` ; and I didn't manage to really fix it. Actually I am thihnking that's just the EDF endpoint which doesn't respond well sometimes ... : /